components/invitation_modal/invitation_modal.tsx

```diff
@@ -189,14 +190,26 @@ export class InvitationModal extends React.PureComponent<Props, State> {
                 invites = result.data;
             }
         } else if (inviteAs === InviteType.GUEST) {
-            const result = await this.props.actions.sendGuestsInvites(
-                this.props.currentTeam.id,
-                this.state.invite.inviteChannels.channels,
-                users,
-                emails,
-                this.state.invite.customMessage.open ? this.state.invite.customMessage.message : '',
-            );
-            invites = result.data;
+            // make sure the endpoint is not called if the guest invitation is restricted by subscription restriction
+            // but somehow the enduser finds the way to manipulate DOM (see MM-47228) to overpass restriction by enabling disabled elements
+            if (this.props.isCloud && !this.props.isCloudFreeTrial && !this.props.isPaidSubscription) {
+                invites.notSent.push({
```

Comment:
Wondering if this change is necessary if we are already checking in the backend?

Reply:
agree, however this looks more like the right implementation of the business logic. Im 2/5 for keeping this.